### ISSUE TITLE
After login it's redirect me to the "redirect_url"

### ISSUE THREAD

**Issue by sameh-serag** [krisrak/jquery-cordova-oauth2#23]
i tried your example. 
function oauth2_login() {
     $.oauth2({
        auth_url: 'https://www.facebook.com/dialog/oauth',
        response_type: 'token',
        client_id: 'XXXXXXXXXXX',
        redirect_uri: 'http://www.zonnix.net',
        other_params: {}
    }, function(token, response){
//        makeAPICalls(token);
         return;
    }, function(error, response){
                 console.log(error);
         return;
    }); 

when i tried it in the emulator ... after successful login the page redirected to the "redirect_url"
Please could u tell me how to avoid the redirect. The facebook force me to insert "redirect_url" related to my valid domain

**Comment by blokeish** [comment]
Yes, with cordova you don't want to be redirected to an external page but return to your app. How can this be done?

**Comment by krisrak** [comment]
@blokeish This plugin is not opening external web browser window, it is opening a webview within the app, so when redirect happens it just closes the webview modal view. Its always in the app.

@sameh-serag you can just put your domain url for redirect url, its not really using it, just grabbing url-param, so does not matter contents of the page

**Comment by blokeish** [comment]
Thanks krisrak, that make sense. I will try it out and see how it goes..

**Comment by joshuaquek** [comment]
Ok, I figured a solution to this.

Firstly, the latest version of Cordova's InAppBrowser doesn't use 
`window.open` anymore.

The fix would be to change line 85 inside cordova.oauth2.js 
from:
`var loginWindow = window.open(login_url, '_blank', 'location=yes,clearsessioncache=yes');`

to 

`var loginWindow = cordova.InAppBrowser.open(login_url, '_blank', 'location=yes,clearsessioncache=yes');`

This should fix the issue.

Also, remember to import in InAppBrowser to your project:
`cordova plugin add cordova-plugin-inappbrowser`

and also, remember to include cordova.js inside your <head> tag:
```
<head>
  <script type="text/javascript" src="cordova.js"></script>
  <script type="text/javascript" src="css/jquery.min.js"></script>
  <script type="text/javascript" src="css/cordova.oauth2.js"></script>
</head>
```


This was the solution that worked for me.
